### PR TITLE
feat(layout): basic mobile and desktop structures

### PR DESCRIPTION
**Pull request by jeysal**

### CODE REVIEW

File: packages/react-components/src/templates/BasicLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
Comment (jeysal) [inline packages/react-components/src/templates/BasicLayout.tsx:1]: This now has essentially zero overlap with the main layout and would only make it even more complex if in the same file, so I extracted it.

File: packages/react-components/src/templates/WelcomePage.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ const containerStyles = css({
 });
 
 const headerStyles = css({
-  flexGrow: 9999,
+  flexGrow: 1,
```
Comment (fampinheiro) [inline packages/react-components/src/templates/WelcomePage.tsx:53]: Thank you

File: packages/react-components/src/templates/Layout.tsx [hunk 3]
```diff
@@ -1,65 +1,179 @@
...
+import { tabletScreen } from '../pixels';
+import { MenuHeader, MainNavigation, UserNavigation } from '../organisms';
+import { UserMenuButton } from '../molecules';
+import { Overlay } from '../atoms';
 
-const showNavigationQuery = `@media (min-width: ${smallDesktopScreen.width}px)`;
+const drawerQuery = `@media (max-width: ${tabletScreen.width - 1}px)`;
+const crossQuery = `@media (min-width: ${tabletScreen.width}px)`;
 
 const styles = css({
```
Comment (jeysal) [inline packages/react-components/src/templates/Layout.tsx:14]: This is about as small as this component can get unfortunately, I've already extracted out everything that isn't tightly coupled to the grid layout

File: packages/react-components/src/templates/WelcomePage.tsx (hunk 2 above)
Comment (jeysal) [reply packages/react-components/src/templates/WelcomePage.tsx:53]: lol, I kinda just moved the hack

Comment (jeysal) [reply packages/react-components/src/templates/WelcomePage.tsx:53]: but it's a fairly common CSS hack :shrug:

**Comment by jeysal** [comment]
This looks like *a lot* to review, but really `Layout` and `Overlay` are new, everything else is basically stub components.

**Review by fampinheiro — APPROVED** [review]

**Review by kmaid — APPROVED** [review]

**Pull request merged**